packages/expo-sqlite/src/Storage.ts

```diff
@@ -0,0 +1,392 @@
+      }
+    });
+  };
+
+  /**
+   * Alias for [`closeAsync()`](#closeasync-1) method.
+   */
+  close = async (): Promise<void> => {
+    this.closeAsync();
+  };
+
+  //#endregion
+
+  //#region Internals
+
+  private async getDbAsync(): Promise<SQLiteDatabase> {
+    if (!this.db) {
+      const db = await openDatabaseAsync(this.databaseName);
+      await this.maybeMigrateDbAsync(db);
+      this.db = db;
+    }
+    return this.db;
+  }
+
+  private getDbSync(): SQLiteDatabase {
+    if (!this.db) {
+      const db = openDatabaseSync(this.databaseName);
+      this.maybeMigrateDbSync(db);
+      this.db = db;
+    }
+    return this.db;
+  }
+
+  private async maybeMigrateDbAsync(db: SQLiteDatabase) {
+    await db.withExclusiveTransactionAsync(async (tx) => {
+      const result = await tx.getFirstAsync<{ user_version: number }>('PRAGMA user_version');
+      let currentDbVersion = result?.user_version ?? 0;
+      if (currentDbVersion >= DATABASE_VERSION) {
+        return;
+      }
+      if (currentDbVersion === 0) {
+        await tx.execAsync(MIGRATION_STATEMENT_0);
+        currentDbVersion = 1;
+      }
+      await tx.execAsync(`PRAGMA user_version = ${DATABASE_VERSION}`);
+    });
+  }
+
+  private maybeMigrateDbSync(db: SQLiteDatabase) {
+    db.withTransactionSync(() => {
+      const result = db.getFirstSync<{ user_version: number }>('PRAGMA user_version');
+      let currentDbVersion = result?.user_version ?? 0;
+      if (currentDbVersion >= DATABASE_VERSION) {
+        return;
+      }
+      if (currentDbVersion === 0) {
+        db.execSync(MIGRATION_STATEMENT_0);
+        currentDbVersion = 1;
+      }
+      db.execSync(`PRAGMA user_version = ${DATABASE_VERSION}`);
+    });
+  }
+
+  /**
+   * Recursively merge two JSON objects.
+   */
+  private static mergeDeep(target: any, source: any): any {
+    if (typeof target !== 'object' || target === null) {
+      return source;
+    }
+
+    if (typeof source !== 'object' || source === null) {
+      return target;
+    }
+
+    const output = { ...target };
+
+    for (const key of Object.keys(source)) {
+      if (source[key] instanceof Array) {
+        if (!output[key]) {
+          output[key] = [];
+        }
+        output[key] = output[key].concat(source[key]);
+      } else if (typeof source[key] === 'object') {
+        output[key] = this.mergeDeep(target[key], source[key]);
+      } else {
+        output[key] = source[key];
+      }
+    }
+
+    return output;
+  }
+
+  //#endregion
+}
+
+/**
+ * This default instance of the [`SQLiteStorage`](#sqlitestorage-1) class is used as a drop-in replacement for the `AsyncStorage` module from [`@react-native-async-storage/async-storage`](https://github.com/react-native-async-storage/async-storage).
+ */
+export const AsyncStorage = new SQLiteStorage('ExpoSQLiteStorage');
+
+export default AsyncStorage;
+
+/**
+ * Alias for [`AsyncStorage`](#asyncstorage-1), given the storage not only offers asynchronous methods.
```

Comment:
```suggestion
 * Alias for [`AsyncStorage`](#sqliteasyncstorage), given the storage not only offers asynchronous methods.
```

Broken internal link.

Reply:
good catch thanks 💛